dask/layers.py

```diff
@@ -844,3 +868,85 @@ def _construct_graph(self, deserializing=False):
             dsk[(self.name, i)] = (concat_func, _concat_list)
 
         return dsk
+
+
+class DataFrameIOLayer(Blockwise, DataFrameLayer):
+    """DataFrame-based Blockwise Layer with IO
+
+    Parameters
+    ----------
+    name : string
+        Name to use for the constructed layer.
+    columns : string, list or None
+        Field name(s) to read in as columns in the output.
+    inputs : list[tuple]
+        List of input-argument tuples. Each element will be
```

Comment:
The "argument" in this case should be a Delayed object, right? Might be nice to say that explicitly here.

Reply:
Each element of `inputs` (a list) will be a tuple of arguments that the IO function should be able to accept in `__call__`.  The element sof the tuple can be anything (paths, row-group indices, kwarg dicts, etc..).

Reply:
Got it, thanks. So though it does happen to be a common case that they're Delayed objects, it's not necessary.